src/core/internal/arrayop.d

```diff
@@ -271,7 +271,7 @@ string scalarExp(Args...)()
         else static if (isUnaryOp(arg))
         {
             auto op = arg[0] == 'u' ? arg[1 .. $] : arg;
-            stack[$ - 1] = op ~ stack[$ - 1];
+            stack[$ - 1] = op ~ "cast(int)"~ stack[$ - 1];
```

Comment:
This is the culprit. Deleting `"cast(int)"~` fixes the problem.

I think there's a missing check that `Args` contains integral types! The cast causes problems when performing array ops on `double[]`.